models/contacts.js

```diff
@@ -1,19 +1,79 @@
-// const fs = require('fs/promises')
+import { promises as fs } from "fs";
+import { nanoid } from "nanoid";
+import path from "path";
+import { schema, schemaReq } from "./validation.js";
+const contactsPath = path.join(process.cwd(), "models", "contacts.json");
 
-const listContacts = async () => {}
+const listContacts = async () => {
+  const contacts = await fs.readFile(contactsPath);
+  return JSON.parse(contacts);
+};
 
-const getContactById = async (contactId) => {}
+const getContactById = async (contactId) => {
+  const contacts = await listContacts();
+  const contact = contacts.find((e) => e.id === contactId);
+  if (!contact) {
+    throw new Error("Not found");
+  }
+  return contact;
+};
 
-const removeContact = async (contactId) => {}
+const removeContact = async (contactId) => {
+  const contacts = await listContacts();
+  const newContacts = contacts.filter((e) => e.id !== contactId);
+  if (contacts.length === newContacts.length) {
+    throw new Error("Not found");
+  }
+  fs.writeFile(contactsPath, JSON.stringify(newContacts, null, 2));
+};
 
-const addContact = async (body) => {}
+const addContact = async (body) => {
+  const { name, email, phone } = body;
+  if (schemaReq.validate({ name, email, phone }).error) {
+    throw new Error(schemaReq.validate({ name, email, phone }).error.message);
+  }
```

Comment:
Niepotrzebnie dwa razy. 
```suggestion
const errMsg = schemaReq.validate({ name, email, phone }).error.message;
  if (errMsg) throw new Error(errMsg);

```